Libplanet/Net/BlockCompletion.cs

```diff
@@ -206,110 +204,35 @@ public bool Satisfy(Block<TAction> block)
-                                            cancellationToken
-                                        );
-                                    }
+                        cancellationToken.ThrowIfCancellationRequested();
+                        IList<HashDigest<SHA256>> hashDigests =
+                            hashes is IList<HashDigest<SHA256>> l ? l : hashes.ToList();
 
-                                    demands.Remove(block.Hash);
-                                }
-                            }
-                            catch (OperationCanceledException e)
-                            {
-                                if (ct.IsCancellationRequested)
-                                {
-                                    _logger.Error(
-                                        e,
-                                        "A blockFetcher job (peer: {Peer}) is cancelled.",
-                                        peer
-                                    );
-                                    throw;
-                                }
+                        cancellationToken.ThrowIfCancellationRequested();
+                        await pool.SpawnAsync(
+                            CreateBatch(hashDigests),
+                            cancellationToken: cancellationToken
+                        );
+                    }
 
-                                _logger.Debug(
-                                    e,
-                                    "Timed out to wait a response from {Peer}.",
-                                    peer
-                                );
-                            }
-                        }
-                        finally
-                        {
-                            if (demands.Any())
-                            {
-                                _logger.Verbose(
-                                    "Fetched blocks from {Peer}, but there are still " +
-                                    "unsatisfied demands ({UnsatisfiedDemandsNumber}) so " +
-                                    "enqueue them again: {UnsatisfiedDemands}.",
-                                    peer,
-                                    demands.Count,
-                                    demands
-                                );
-                                Demand(demands, retry: true);
-                            }
-                            else
-                            {
-                                _logger.Verbose("Fetched blocks from {Peer}.", peer);
-                            }
-                        }
-                    },
-                    cancellationToken: cancellationToken
-                );
-            }
+                    await pool.WaitAll();
+                }
+                finally
+                {
+                    queue.CompleteAdding();
+                }
+            });
+#pragma warning restore CS4014 // It's okay because we'll wait at next block.
```

Comment:
Instead of suppressing CS4014, how about storing the returning value somewhere at one point, and then `await`ing it in the very last of this method?